### PR TITLE
Enable path-based interactive tools

### PR DESCRIPTION
**Pull request by kysrpex**
Enable access via path (e.g. `https://usegalaxy.eu/interactivetool/ep/3qg8nstfnl44r/fsjbqy66kln5`) to interactive tools that support it (those for which `requires_domain="False"` in the tool's xml file).

More information:
- nginx configuration: https://docs.galaxyproject.org/en/master/admin/special_topics/interactivetools.html#nginx-proxy-server-configuration-in-production
- interactive tools proxy configuration: https://github.com/galaxyproject/galaxy/pull/18481#issuecomment-2217313953, https://github.com/galaxyproject/gx-it-proxy/blob/v0.0.6/lib/proxy.js#L101-L141, https://github.com/galaxyproject/gx-it-proxy/blob/v0.0.6/lib/main.js#L16

Related commit: [usegalaxy-eu/galaxy@e216b45](https://github.com/usegalaxy-eu/galaxy/commit/e216b4549be02a12c16d50313935ad0dfa073b90#comments)

### CODE REVIEW

**Comment by kysrpex** [comment]
Ping @sveinugu (just FYI)

**Comment by sveinugu** [comment]
Great!

Could you also update the ansible playbook example you mentioned?

File: templates/nginx/galaxy-main.j2 [hunk 1]
```diff
@@ -181,6 +181,16 @@ server {
...
 
+	# Route all path-based interactive tool requests to the InteractiveTool proxy application
+    location ~* ^/(interactivetool/.+)$ {
+        proxy_redirect off;
+        proxy_http_version 1.1;
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection "upgrade";
+        proxy_pass http://127.0.0.1:{{ gie_proxy_port }};
```
Comment (sanjaysrikakulam) [inline templates/nginx/galaxy-main.j2:192]: I have a question: I see on the main nginx conf that the path `/gie_proxy` is suffixed after the port [here](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/master/templates/nginx/galaxy-main.j2#L171). Is that necessary for this `proxy_pass` as well?

**Comment by bgruening** [comment]
Interesting! Thanks @kysrpex! If this gets merged I will remove my commit in our Galaxy fork and we can try.

File: templates/nginx/galaxy-main.j2 (hunk 1 above)
Comment (kysrpex) [reply templates/nginx/galaxy-main.j2:192]: Actually, I do not know what the path

```
	# Global GIE configuration
	location /gie_proxy {
		proxy_pass http://127.0.0.1:8800/gie_proxy;
		proxy_redirect off;
	}
```

does (maybe @hexylena has a clue? <- git blame). As far as I know, the configuration that is making domain-based interactive tools work is [this one](https://github.com/usegalaxy-eu/infrastructure-playbook/blob/fce2056fbe7f06245dd47bb453a618c832de2602/templates/nginx/galaxy-ie-main.j2#L14-L26).

```
	server_name ~^(?<key>[0-9a-z-]*)\.ep\.interactivetool\.(?<domain>[a-z0-9.-]*)usegalaxy\.eu$;

	[...]

	location / {
		proxy_pass http://127.0.0.1:8800;
		proxy_redirect off;
		proxy_http_version 1.1;
		proxy_set_header Host $host;
		proxy_set_header Upgrade $http_upgrade;
		proxy_set_header Connection $connection_upgrade;
	}
```

See also the [Galaxy docs](https://docs.galaxyproject.org/en/latest/admin/special_topics/interactivetools.html#nginx-proxy-server-configuration-in-production). Both in the excerpt above and the Galaxy docs you can see that it is not necessary to add the prefix again. I assume it's needed only if you want special behavior (e.g. alter the prefix).

**Comment by kysrpex** [comment]
> Great!
> 
> Could you also update the ansible playbook example you mentioned?

@sveinugu Commit https://github.com/usegalaxy-eu/infrastructure-playbook/pull/1256/commits/a97941bff0e9d032eae7f3e807d5c36bcadeab31 would not be valid without having updated the Ansible role, see the [latest release from 2h ago](https://github.com/usegalaxy-eu/ansible-gie-proxy/releases/tag/0.1.0) and the [comment](https://github.com/usegalaxy-eu/ansible-gie-proxy/blob/615a51f9fc430cd96f95d67fa3e8fdb1361d5f43/README.md?plain=1#L41-L43) on the [new README](https://github.com/usegalaxy-eu/ansible-gie-proxy/blob/615a51f9fc430cd96f95d67fa3e8fdb1361d5f43/README.md).

>     # configuring this path prefix enables path-based interactive tools
>     # (https://docs.galaxyproject.org/en/master/admin/special_topics/interactivetools.html#nginx-proxy-server-configuration-in-production)
>     gie_proxy_path_prefix: /interactivetool/ep

File: templates/nginx/galaxy-main.j2 (hunk 1 above)
Comment (sveinugu) [reply templates/nginx/galaxy-main.j2:192]: True. Adding an extra prefix would OTOH mess up path-based ITs. 

Also the solution for supporting path-based ITs might differ from tool to tool. The config @sanjaysrikakulam refers to above will work if all paths in the web server in the IT are relative. If not, other solutions for injecting the path are available, as
documented in the Galaxy admin docs.

**Pull request merged**